common/src/main/java/com/wynntils/features/ui/CustomLoadingScreenFeature.java

```diff
@@ -12,29 +12,45 @@
 import com.wynntils.mc.event.ResourcePackEvent;
 import com.wynntils.mc.event.ScreenOpenedEvent;
 import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.mc.event.TickAlwaysEvent;
 import com.wynntils.mc.event.TitleSetTextEvent;
 import com.wynntils.models.worlds.event.WorldStateEvent;
 import com.wynntils.screens.characterselector.LoadingScreen;
 import com.wynntils.utils.mc.McUtils;
+import net.minecraft.client.gui.screens.ConnectScreen;
+import net.minecraft.client.gui.screens.DisconnectedScreen;
 import net.minecraft.client.gui.screens.ProgressScreen;
 import net.minecraft.client.gui.screens.ReceivingLevelScreen;
+import net.minecraft.client.resources.language.I18n;
 import net.minecraftforge.eventbus.api.EventPriority;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 @ConfigCategory(Category.UI)
 public class CustomLoadingScreenFeature extends Feature {
     private LoadingScreen loadingScreen;
+    private ConnectScreen baseConnectScreen;
+    // Minecraft does some of it's connection logic in ConnectScreen which is strange
+    // We need to be able to tell our custom loading screen to work with it in the background
 
-    @SubscribeEvent(priority = EventPriority.HIGH)
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
     public void onScreenOpenPre(ScreenOpenedEvent.Pre event) {
+        if (event.getScreen() instanceof ConnectScreen cs) {
+            baseConnectScreen = cs;
+        }
+
         if (loadingScreen == null) return;
 
         if (event.getScreen() instanceof ProgressScreen) {
             event.setCanceled(true);
+            loadingScreen.setMessage(I18n.get("feature.wynntils.customLoadingScreen.working"));
         }
         if (event.getScreen() instanceof ReceivingLevelScreen) {
             event.setCanceled(true);
-            loadingScreen.setMessage("Receiving terrain...");
+            loadingScreen.setMessage(I18n.get("feature.wynntils.customLoadingScreen.receivingTerrain"));
+        }
+
+        if (event.getScreen() instanceof DisconnectedScreen) {
```

Comment:
This does not seem like the correct condition to set baseConnectScreen to null...? Or at least it should also be set to null when the ConnectScreen closes.

Reply:
There is no actual way to check if the ConnectScreen closes, because, well, we got rid of it and have our own. But I think what I have here is fine. Any screen that isn't our custom screen or some un-cancelled ConnectScreen will erase it. This allows it to work during our connection, and closes it as soon as the handshake is done (including ProgressScreen, ReceivingLevelScreen, etc), it'll disappear nicely.

Reply:
Ok. Sounds good then. Maybe add a shortened version of this explanation as a comment?